best-first-cv/script.whizzml

```diff
@@ -0,0 +1,561 @@
+                                          "replacement"
+                                          "randomize"
+                                          "seed"]))
+(define LOGISTIC_OPTIONS ["balance_fields"
+                          "input_fields"
+                          "bias"
+                          "c"
+                          "missing_numerics"
+                          "default_numeric_value"
+                          "eps"
+                          "field_codings"
+                          "normalize"
+                          "regularization"
+                          "seed"])
+(define EVALUATION_OPTIONS ["sample_rate"
+                            "input_fields"
+                            "out_of_bag"
+                            "range"
+                            "replacement"
+                            "ordering"
+                            "seed"
+                            "missing_strategy"
+                            "combiner"])
+;; cross-validation
+;;
+;; creates k-fold cross-validation for a dataset
+;; Inputs:
+;;   dataset-id: (string) Dataset ID
+;;   k-folds: (integer) Number of folds
+;;   model-options: (map) Options to use in model/ensemble
+;;   evaluation-options: (map) Options to use in evaluation creation
+;;   delete-resources: (boolean) Whether to delete all intermediate resources
+;;
+;; Output: (map) Average of evaluations results
+;;
+;; Raises:
+;;  101: The dataset-id argument is not a string
+;;  102: The dataset-id is not a valid dataset ID
+;;  103: The k-folds argument is not an integer
+;;  104: The k-folds argument is not >= 2
+;;  106: The objective field ID is not in the selectable IDs list
+;;
+(define (cross-validation k-fold-datasets
+                          objective-id
+                          model-options
+                          evaluation-options
+                          delete-resources)
+  (let (dataset (fetch (k-fold-datasets 0))
+        dataset-name (dataset "name" false))
+    (check-dataset-objective-id objective-id dataset)
+    (let (objective-name (get-objective-name dataset objective-id)
+          evaluations (create-k-evaluations k-fold-datasets
+                                            objective-name
+                                            dataset-name
+                                            model-options
+                                            evaluation-options
+                                            delete-resources)
+          ;; commented out till wintermute's new version
+          ;; evaluations-average (create-and-wait-evaluation {"evaluations" evaluations})
+          average-phi (get-average-phi evaluations)
+          _ (log-info "model options: " model-options)
+          _ (log-info "average phi: " average-phi))
+      (when delete-resources
+        (map safe-delete evaluations))
+      average-phi)))
+
+
+(define (safe-create type args)
+  (try
+    (create type args)
+    (catch e
+      (create type args))))
```

Comment:
i think that most of the time, creating resources fails after a while, so the call to `create` almost never throws an error... it's only later, when `wait` is called (perhaps implicitly) that the exception is raised.  so without explicitly waiting this function is probably not that useful...